src/QueryPressure.Mongodb.Core/MongodbConnectionProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿using MongoDB.Driver;
+using QueryPressure.Core.Interfaces;
+using QueryPressure.Core.Requirements;
+
+namespace QueryPressure.Mongodb.Core;
+
+public class MongodbConnectionProvider : IConnectionProvider
+{
+  private readonly string _connectionString;
+  private readonly string _collectionName;
+
+  public MongodbConnectionProvider(string connectionString, string collectionName)
+  {
+    _connectionString = connectionString;
+    _collectionName = collectionName;
+  }
+
+  public async Task<IExecutable> CreateExecutorAsync(IScriptSource scriptSource, ConnectionRequirement connectionRequirement,
+    CancellationToken cancellationToken)
+  {
+    var script = await scriptSource.GetScriptAsync(cancellationToken);
+    
+    var connection = new MongoClient(_connectionString);
```

Comment:
How it will work with concurrent queries?

Reply:
MongoClient is thread safe to the documentation.
http://mongodb.github.io/mongo-csharp-driver/2.0/reference/driver/connecting/